src/senaite/sync/syncstep.py

```diff
@@ -194,16 +194,19 @@ def _fetch_missing_parents(self, item):
         to fill the missing parents for fetched objects.
         :return:
         """
+        # Never fetch parents of an unnecessary objects
+        if not utils.is_item_allowed(item):
+            return False
         parent_path = item.get("parent_path")
         # Skip if the parent is portal object
         if self.is_portal_path(parent_path):
-            return
+            return True
```

Comment:
Please take care about class/method/parameter namings.
If you return True/False, then your method should have a name like `is_` or `check_`, if it fetches an attribute, then it should have a name like `get_` or `fetch_` etc.
In this case I would expect a list of parents, but get `True` or `False` and I have no idea what these values mean.
IMO also these `_` methods are a pain in the eyes, because they look like a regular expression.

Reply:
Sorry but I didn't understand which `_` functions you are referring to.

Reply:
Something like `_fetch_data `, `_fetch_settings`, `_get_settings_by_key`, `_fetch_registry_records` ... I think it is not necessary to mark them as private as long as they do not change the state of the whole class (remember the functional and immutable programming approach).
However, it is just a style issue of course, but we should put readability and maintainability of the code before any other paradigms.

Always keep in mind that "Code Speaks" – if the function name + parameters are self explaining and the code has no side effects, e.g. do no change some `self.xxx` attributes, you do not need to write exhaustive documentation about what the function/method do, because it is obvious what it does.

I hope this makes sense to you, because I do not dictate my ideology ony you;)

Reply:
Obviously it does, I really like, and want you to correct me, I want to be better on Python : )
About those funcitons, I thought `_` is an _obligatory_ indicator of `private` or `protected` methods which you do not want to be called from everywhere. Actually, most of those classes would fail if you just create an instance of their class and call them. They are like dependent on each other. So, to be safe I mark them with `_`.

Reply:
Agree that, as per the name of this function (`fetch_*`), the first thing that comes in my mind is that the function will return a list. Please change the name to something like `is_missing_parents`